be/src/olap/in_list_predicate.h

```diff
@@ -241,6 +242,29 @@ class InListPredicateBase : public ColumnPredicate {
         return Status::OK();
     }
 
+    Status evaluate(const Schema& schema, InvertedIndexIterator* iterator,
+                    uint32_t num_rows, roaring::Roaring* result) const override {
+        if (iterator == nullptr) {
+            return Status::OK();
+        }
+        auto column_desc = schema.column(_column_id);
+        std::string column_name = column_desc->name();
+        roaring::Roaring indices;
+        for (auto value : *_values) {
+            InvertedIndexQueryType query_type = InvertedIndexQueryType::EQUAL_QUERY;
+            roaring::Roaring index;
+            RETURN_IF_ERROR(
+                    iterator->read_from_inverted_index(column_name, &value, query_type, num_rows, &index));
+            indices |= index;
+        }
+        if constexpr (PT == PredicateType::IN_LIST) {
+            *result &= indices;
+        } else {
+            *result -= indices;
```

Comment:
should check NOT_IN_LIST explicitly

Reply:
no need, because only IN_LIST and NOT_IN_LIST predicate will be execute by InListPredicateBase